extensions/wikia/FeedsAndPosts/JustUpdatedAPIProxy.class.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Wikia\FeedsAndPosts;
+
+class JustUpdatedAPIProxy {
+
+	const LIMIT = 4;
+
+	private function requestAPI($params) {
+		$api = new \ApiMain(new \FauxRequest($params));
+		$api->execute();
+
+		return $api->GetResultData();
+	}
+
+	private function getRecentChanges() {
+		$response = $this->requestAPI( [
+			'action' => 'query',
+			'list' => 'recentchanges',
+			'rcprop' => 'title',
+			'rcshow' => '!bot|!minor|!redirect',
+			'rcnamespace' => '0',
+			'rclimit' => self::LIMIT,
+		] );
+
+		if ( isset( $response['query']['recentchanges'] ) ) {
+			return $response['query']['recentchanges'];
+		}
+
+		return [];
+	}
+
+	private function getImage($title) {
+		$response = $this->requestAPI( [
+			'action' => 'imageserving',
+			'wisTitle' => $title,
+		] );
+
+		if ( isset( $response['image']['imageserving'] ) ) {
+			return $response['image']['imageserving'];
+		}
+
+		return null;
+	}
+
+	public function get() {
+		return \WikiaDataAccess::cacheWithOptions(
+			wfMemcKey( 'feeds-just-updated' ),
+			function () {
+				$recentChanges = $this->getRecentChanges();
+
+				foreach($recentChanges as &$article) {
+					$article['image'] = $this->getImage($article['title']);
+				}
+
+				return $recentChanges;
+			},
+			[
+				'command' => \WikiaDataAccess::USE_CACHE,
+				'cacheTTL' => \WikiaResponse::CACHE_VERY_SHORT,
```

Comment:
according to ticket it can be cached up to an hour, why do we use 5 min instead?